.github/workflows/pull_requests.yaml

```diff
@@ -10,66 +10,25 @@ jobs:
         os: ["ubuntu-latest", "windows-latest", "macos-latest"]
     runs-on: ${{ matrix.os }}
     steps:
+      - name: Install Bash 4 on Mac
+        if: matrix.os == 'macos-latest'
+        run: |
+          /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+          brew update
+
+          brew install bash
+          brew install gnu-sed
```

Comment:
By requiring this update from the Mac-default BSD `sed` to `gnu-sed`, the `make` command will fail by default for mac users (developers) with a fairly cryptic message.  See the suggestion to the script below for an alternate solution that works on both platforms.